Neos.Media/Classes/Domain/Model/Asset.php

```diff
@@ -412,6 +445,56 @@ public function setAssetCollections(Collection $assetCollections)
         $this->assetCollections = $assetCollections;
     }
 
+
+    /**
+     * @param string $assetSourceIdentifier
+     */
+    public function setAssetSourceIdentifier(string $assetSourceIdentifier): void
+    {
+        $this->assetSourceIdentifier = $assetSourceIdentifier;
+    }
+
+    /**
+     * @return string
+     */
+    public function getAssetSourceIdentifier(): string
+    {
+        return $this->assetSourceIdentifier;
+    }
+
+    public function getAssetSource(): ?AssetSource
```

Comment:
Does the Asset actually need this? And shouldn't the resource be instantiated lazily in here?

Reply:
No, that's not needed (anymore), I'll refactor this, also to be initialized lazily.